### DISCUSSION TITLE
Plugin not working on Netbox 3.3.7

### DISCUSSION THREAD

**Discussion by emintjeee** [mattieserver/netbox-topology-views#219] [Q&A]
Topology Views doesnt view anything after updating everything, 
![Screenshot_3](https://user-images.githubusercontent.com/95137379/200532778-48339580-6182-4ca5-8d9e-be647300cbd3.png)

**Comment by cyberndj** [comment]
Howdy,
By chance, have your run `python3 manage.py collectstatic --no-input` after the upgrade?

**Comment by emintjeee** [comment]
After installing all the pip modules required at manage.py it works again, only i get this error, littlebit strange because the plugin works fine now. any solution?
![image](https://user-images.githubusercontent.com/95137379/200775719-3aeb68f2-db1f-4fc1-b956-901dbef707c8.png)

**Comment by mattieserver** [comment]
When you install the requirements, is this with the venv active?
________________________________

**Comment by thr3swe** [comment]
I have the same issue and I've run "python3 manage.py collectstatic --no-input" with the environment activated. Also did pip3 install -r requirements.txt and rebooted the server. 
![image](https://user-images.githubusercontent.com/35501825/201089070-9b0b5e88-db5a-4581-aebe-6e4f36e2478b.png)
![image](https://user-images.githubusercontent.com/35501825/201089545-40e0dac4-4a80-43d9-9144-900d0c1a870f.png)

**Comment by emintjeee** [comment]
Yes i did, now i got the same as thr3swe.

**Comment by mattieserver** [comment]
hi,

I will check if can reproduce this tomorrow.
Currently i have not tested the plugin on 3.3.7 of netbox.

**Comment by cernikradek** [comment]
Yes, I have the same problem on Netbox version 3.3.7. Thank you in advance for solving this problem mattieserver.

**Comment by cernikradek** [comment]
Greetings.
So the plugin works in Netbox version 3.3.7!
What did I do wrong?
I ran "python3 manage.py collectstatic" outside the virtual environment!

It worked for me...
1) source /opt/netbox/venv/bin/activate
2) pip install netbox-topology-views
3) python3 manage.py collectstatic
4) exit
5) sudo echo netbox-topology-views >> local_requirements.txt
6) sudo systemctl restart netbox netbox-rq

Schoolboy mistake.

**Comment by thr3swe** [comment]
I ran inside the virtual environment. 

Went through all your steps now again just to be sure. Didn't solve the issue for me.

**Comment by chrisintheshell** [comment]
I seem to be having a similar issue with **Netbox 3.3.8**. I followed the install instructions to a tee under the venv but I still just get a black screen under the Topology plugin. When I ran `python3 manage.py collectstatic --no-input` I got the following output which I'm assuming is correct:

```
(venv) [root@victa01-netbox01 netbox]# python3 manage.py collectstatic --no-input

22 static files copied to '/opt/netbox-3.3.8/netbox/static', 520 unmodified.
```

UPDATE: Nevermind, it appears to be working. I just needed to search for a site using the filter.

**Comment by quantum909** [comment]
I also have problems in netbox 3.3.8 to display the network topolgy with Netbox in a docker container. After creating the docker container topology view doesnt view anything.

When I try to run 'python3 manage.py collectstatic --no-input' in the active venv, I get the following permission error.

<img width="1156" alt="1" src="https://user-images.githubusercontent.com/5301445/202873343-24a2d0bf-7606-41c9-86c6-0377ce614aeb.png">

<img width="1311" alt="2" src="https://user-images.githubusercontent.com/5301445/202873348-b4efa5ea-5034-4d9c-8886-303dc874b900.png">

**Comment by quantum909** [comment]
It is running now. I had forgotten to add these lines to the dockerfile plugins file.

```
# These lines are only required if your plugin has its own static files.
COPY configuration/configuration.py /etc/netbox/config/configuration.py
RUN SECRET_KEY="dummy" /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py collectstatic --no-input
```
